layouts/market.vue

```diff
@@ -0,0 +1,185 @@
+            </div>
+          </transition>
+        </div>
+        <div class="col-span-6 lg:col-span-9 4xl:col-span-9">
+          <div class="flex flex-wrap flex-col w-full h-full">
+            <div class="w-full">
+              <v-card tight>
+                <div class="grid grid-cols-6 lg:grid-cols-12">
+                  <div class="col-span-6 lg:col-span-8 4xl:col-span-9">
+                    <slot name="chart" />
+                  </div>
+                  <div class="col-span-6 lg:col-span-4 4xl:col-span-3">
+                    <slot name="order-books" />
+                  </div>
+                </div>
+              </v-card>
+            </div>
+            <div class="w-full flex-1 mt-1">
+              <slot name="orders" />
+            </div>
+          </div>
+        </div>
+      </div>
+      <slot name="modals" />
+      <v-modal-market-beta v-if="marketIsBeta" />
+    </div>
+  </VHocLoading>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import { Status, StatusType } from '@injectivelabs/utils'
+import {
+  UiDerivativeMarketWithToken,
+  UiSpotMarketWithToken
+} from '@injectivelabs/ui-common'
+import VSpotMarket from '~/components/partials/spot/market.vue'
+import VDerivativeMarket from '~/components/partials/derivatives/market.vue'
+import VMarketSelection from '~/components/partials/common/markets/markets-selection.vue'
+import VModalMarketBeta from '~/components/partials/modals/market-beta.vue'
+import {
+  ETH_COIN_GECKO_ID,
+  USDT_COIN_GECKO_ID,
+  UST_COIN_GECKO_ID
+} from '~/app/utils/constants'
+import { betaMarketSlugs } from '~/app/data/market'
+import { Modal } from '~/types'
+
+export default Vue.extend({
+  components: {
+    VDerivativeMarket,
+    VMarketSelection,
+    VModalMarketBeta,
+    VSpotMarket
+  },
+
+  props: {
+    overwriteSlug: {
+      type: String,
+      default: undefined
+    }
+  },
+
+  data() {
+    return {
+      status: new Status(StatusType.Loading),
+      isSpotMarket: false,
+      showMarketList: false,
+      interval: 0 as any
+    }
+  },
+
+  computed: {
+    derivativeMarket(): UiDerivativeMarketWithToken | undefined {
+      return this.$accessor.derivatives.market
+    },
+
+    spotMarket(): UiSpotMarketWithToken | undefined {
+      return this.$accessor.spot.market
+    },
+
+    market(): UiSpotMarketWithToken | UiDerivativeMarketWithToken | undefined {
+      const { derivativeMarket, isSpotMarket, spotMarket } = this
+
+      return isSpotMarket ? spotMarket : derivativeMarket
+    },
+
+    slug(): string {
+      const { isSpotMarket, overwriteSlug } = this
+      const { params } = this.$route
+
+      if (overwriteSlug) {
+        return overwriteSlug
+      }
+
+      return isSpotMarket ? params.spot : params.derivative
+    },
+
+    marketIsBeta(): boolean {
+      const { slug } = this
+
+      return betaMarketSlugs.includes(slug)
+    }
+  },
+
+  mounted() {
+    this.isSpotMarket = this.$route.name === 'spot-spot'
```

Comment:
Instead of having this as a `data` property it should be a `computed` property based on the route name.

Reply:
Good point, updated.